src/generic/GenericTypes.jl

```diff
@@ -1038,17 +1038,22 @@ const NCRingElement = Union{NCRingElem, RingElement}
 abstract type Mat{T} <: MatElem{T} end
 
 # not really a mathematical ring
-struct MatSpace{T <: NCRingElement} <: AbstractAlgebra.MatSpace{T}
+struct MatSpace{CoeffT, RingT} <: AbstractAlgebra.MatSpace{CoeffT}
```

Comment:
Strictly speaking the introduction of the second type parameter `RingT` is not necessary for this PR, it just improves type stability, and I saw no harm in doing it?

I dropped the `<: NCRingElement` because `AbstractAlgebra.MatSpace{T}` already enforces it

Reply:
Why would it make everything slow? Note that `CoeffT` uniquely determines `RingT` and vice versa, so it's not as if we'd be creating additional specializations, would we? What am I missing?

Reply:
Sorry, I should have written more. I was thinking of things like
```
x = Vector{MatSpace{T}}()
```
and
```
struct A
  x::MatSpace{T}
```
Do you also change it for the element? I don't understand why we would want to change this all of a sudden (in this PR here).

Reply:
I agree that this shouldn't be in this PR (if we do it at all). I've update the PR accordingly.